deps/childprocess.lua

```diff
@@ -164,4 +164,71 @@ local function spawn(command, args, options)
   return em
 end
 
+local function exec(command, args, options, callback)
+  local file
+  local stringArgs = command..table.concat(args, ' ')
+  if process.env._system_name:lower() == 'win32' then
+    file = 'cmd.exe'
+    args = {'/s', '/c', '"'..stringArgs..'"' }
+  else
+    file = '/bin/sh'
+    args = {'-c', stringArgs }
+  end
+
+  if options.shell then file = options.shell end
+  if not options.maxBuffer then options.maxBuffer = 200 * 1024 end
+
+  -- http://wowwiki.wikia.com/wiki/Lua_object_memory_sizes
+  -- We're being generous here since the internal representation is to chunk them into a table
+  -- which have a base size of 40bytes but the final callback recieves a single string of each type
+  local stdoutchunkbytes = 24
+  local stderrchunkbytes = 24
```

Comment:
Why are we trying to take table size into account?  I would be very confused if I had set a max buffer size of 100 bytes and only got back 76 bytes.